pynumaflow/function/multiproc_server.py

```diff
@@ -0,0 +1,96 @@
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import contextlib
+import os
+from concurrent import futures
+import datetime
+import logging
+import multiprocessing
+import socket
+import time
+import grpc
+
+from pynumaflow._constants import MULTIPROC_FUNCTION_SOCK_PATH, MULTIPROC_FUNCTION_SOCK_ADDR
+from pynumaflow.function.proto import udfunction_pb2_grpc
+
+from pynumaflow import setup_logging
+
+_LOGGER = setup_logging(__name__)
+if os.getenv("PYTHONDEBUG"):
+    _LOGGER.setLevel(logging.DEBUG)
+
+_ONE_DAY = datetime.timedelta(days=1)
+
+
+class MultiProcServer:
+    """
+    Provides a multiprocessing implementation of the grpc server.
+    They use a given TCP socket with SOCK_REUSE to allow all servers,
+    to listen on the same port. We start servers equal to the CPU count of the
+    system.
+    """
+
+    def __init__(self, udf_service, server_options):
```

Comment:
Let's add typing

Reply:
For adding typing here I will need to add a cyclic import, 
is is fine to type check using this then,
`# some_file.py

from __future__ import annotations
from typing import TYPE_CHECKING
if TYPE_CHECKING:
    from main import Main

class MyObject(object):
    # Hooray, cleaner annotations!
    def func2(self, some_param: Main):
        ...`